### DISCUSSION TITLE
Display Folder Tree on Roles & Permissions Page

### DISCUSSION THREAD

**Discussion by raphaelbernhart** [directus/directus#10689] [Feature Requests]
Hello there,

as already fixed in [Issue](https://github.com/directus/directus/issues/9207#event-5669642512) there was a problem with the Permissions with folders. :)

There is just one small problem with the solution except that a long-term fix would be to create a permissions system:

In my case(agency) I have multiple clients(folders) with multiple collections. If the names of the data models overlap there is no chance to see what collection belongs to which customers(folder) in the Roles & Permissions Page. In my case I have two cu
stomers which offer some services. These are managed in collections named services (for every customer one).
On the Roles & Permissions page the folders are not displayed so there are just two Collections named Services.

<img width="1781" alt="Bildschirmfoto 2021-12-25 um 22 59 57" src="https://user-images.githubusercontent.com/48283236/147394255-57694599-6d10-44e8-8e7a-4789bd7503f6.png">
<img width="1070" alt="Bildschirmfoto 2021-12-25 um 22 56 28" src="https://user-images.githubusercontent.com/48283236/147394209-42670a23-1bb4-44e6-8fd5-97f6c3e54741.png">

Atm I name the collections as followed `CUSTOMER_services` but this is not very efficient and/or fancy :/

**Comment by rijkvanzanten** [comment]
I think we should use the technical keys in this table rather than the prettified translated name (@benhaynes). ~~Also, i'm 95% sure I just mentioned that on another issue/discussion, but can't seem to find it now 🙈~~ [There it is!](https://github.com/directus/directus/discussions/9906#discussioncomment-1868737)

**Reply by benhaynes** [reply]
Yeah, my same comment applies... as I'd like to keep this clear, but also friendly. We need to keep the no-code non-technical users in mind for these system areas too. The main goal is to avoid ambiguity.

> Yes. Permissions are technical enough to warrant the actual collection key. Let's keep it with sans-serif font and title format though, and show the actual name (unformatted) and in monospace within a tooltip of the title.

**Reply by raphaelbernhart** [reply]
First of all...thanks for the reply!

Wouldn't it be a simpler solution to just show the folder structure as on the Content Route? :
<img width="216" alt="Bildschirmfoto 2021-12-27 um 18 40 04" src="https://user-images.githubusercontent.com/48283236/147495048-a80ff795-031f-44dd-9528-6a1bc4829793.png">

This shouldn't be a great problem as the structure is already shown on the Content Route.

**Reply by rijkvanzanten** [reply]
@raphaelbernhart It could, though it might cause conflicts with the controls. If you shift the whole row over, you lose that table alignment 🤔

**Reply by LasseRosenow** [reply]
> @raphaelbernhart It could, though it might cause conflicts with the controls. If you shift the whole row over, you lose that table alignment 🤔

@rijkvanzanten 

This could be fixed if we only show the nesting by changing the place of the title like so:

![image](https://user-images.githubusercontent.com/10547444/163292580-7a774033-e1bf-4ca6-892e-ab349d6b26b9.png)

What I would like more though would be to just use the `Data Model` layout, as this would already be familiar to the developer and just show the controls on the very right side of each item. Also Applying the specified items at the left, and if collections are hidden etc. also help the developer to find what they are looking for even faster:

![image](https://user-images.githubusercontent.com/10547444/163294421-a705b90e-39c6-4325-9942-00eac46e9f82.png)

**Reply by LasseRosenow** [reply]
My initial idea was to put in those "-" icons to show that something can not be configured because it's a folder but maybe it would be less confusing to just hide those and make it empty for folders because otherwise one might think the "-" is some special config state or something like that

**Reply by benhaynes** [reply]
I love the shortcuts this would enable. 👍

**Reply by LasseRosenow** [reply]
> I love the shortcuts this would enable. 👍

I had too much free time, so I just gave it a shot: #12774 I hope you like it

**Reply by LasseRosenow** [reply]
Based on the feedback in my PR #12774 I made a new proposal how to solve this. Maybe should put it here also:

To revisit this, I thought a little more about this and maybe the best way for this would be to show the internal name by default like in the `Model` page.

So in general my new Proposal based on all the feedback is this:
- Regarding the Tree-View:
  - Don't add tree-view based on the `Model` page structure
  - (Enable Tree-View in a later PR (but still unrelated to the `Model` page structure) with the ability to create `Folders/Groups` that can have rules that would be valid for all collections contained in them. (That could help people safe a lot of time setting up the same permissions for multiple collections))?
- Sort items the same way as in the `Model` page
- Show internal names by default instead of the translation just like in the `Model` page

**Reply by LasseRosenow** [reply]
Would like to hear the opinion of the directus team regarding if they would still prefer the tree structure rather than my new proposal or not. Because they didn't mention in the PR. I know you all are very busy, so no problem for me haha :)

In general I would be fine with both solutions, but we really need to overcome the current situation, which causes so many problems :D

**Reply by rijkvanzanten** [reply]
@LasseRosenow I also posted this on the PR, but TL;DR: I love the design of the tree-view, _but_ i'm a bit worried that it's too confusing, as it feels like the nested stuff would inherit the permissions of the parent. I think Gerard's idea of showing a breadcrumb per row is a good middleground 🙂 

> I know you all are very busy, so no problem for me haha :)

hey psst come help me be slightly less busy so we can focus on things like this 👉🏻 https://directus.io/careers

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
